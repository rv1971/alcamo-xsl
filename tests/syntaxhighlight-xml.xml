<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns="http://www.w3.org/1999/xhtml"
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#"
      xmlns:sh="tag:rv1971@web.de,2021:alcamo-xsl:syntaxhighlight:xml#"
      xml:id="xslt"
      version="1.0"
      exclude-result-prefixes="a sh">
    <xsl:import href="../xsl/syntaxhighlight-xml.xsl"/>

    <xsl:template match="/">
      <pre>
        <xsl:apply-templates mode="sh:xml"/>
     </pre>
    </xsl:template>

    <xsl:template name="foo">
      <xsl:text>This text
will not
be indented.</xsl:text>
    </xsl:template>
  </xsl:stylesheet>

  <!-- comment -->
  <foo bar="long value which makes the total length exceed 60 characters"/>

  <foo>Short inline text.</foo>

  <foo>Lorem ipsum dolor sit amet, consetetur sadipscing elitr.</foo>

  <bar content="elements without child nodes are not surrounded by empty lines"/>
  <baz content="elements without child nodes are not surrounded by empty lines"/>
</a:doc>
