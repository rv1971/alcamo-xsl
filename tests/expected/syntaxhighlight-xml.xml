<?xml version="1.0"?>
<pre xmlns="http://www.w3.org/1999/xhtml">&lt;<span class="sh-prefix">a</span>:<span class="sh-element">doc</span>
    <span class="sh-xmlns-prefix">xmlns</span>:<span class="sh-prefix">xml</span>=<span class="sh-namespace-uri">"<a href="http://www.w3.org/XML/1998/namespace">http://www.w3.org/XML/1998/namespace</a>"</span>
    <span class="sh-xmlns-prefix">xmlns</span>:<span class="sh-prefix">a</span>=<span class="sh-namespace-uri">"tag:rv1971@web.de,2021:alcamo-xsl#"</span>
    <span class="sh-xmlns-prefix">xmlns</span>:<span class="sh-prefix">xsd</span>=<span class="sh-namespace-uri">"<a href="http://www.w3.org/2001/XMLSchema">http://www.w3.org/2001/XMLSchema</a>"</span>&gt;
  &lt;<span class="sh-prefix">xsl</span>:<span class="sh-element">stylesheet</span>
      <span class="sh-xmlns-prefix">xmlns</span>:<span class="sh-prefix">sh</span>=<span class="sh-namespace-uri">"tag:rv1971@web.de,2021:alcamo-xsl:syntaxhighlight:xml#"</span>
      <span class="sh-xmlns-prefix">xmlns</span>:<span class="sh-prefix">xsl</span>=<span class="sh-namespace-uri">"<a href="http://www.w3.org/1999/XSL/Transform">http://www.w3.org/1999/XSL/Transform</a>"</span>
      <span class="sh-xmlns-prefix">xmlns</span>=<span class="sh-namespace-uri">"<a href="http://www.w3.org/1999/xhtml">http://www.w3.org/1999/xhtml</a>"</span>
      <span class="sh-prefix">xml</span>:<span class="sh-attr">id</span>=<span class="sh-attr-value">"xslt"</span>
      <span class="sh-attr">version</span>=<span class="sh-attr-value">"1.0"</span>
      <span class="sh-attr">exclude-result-prefixes</span>=<span class="sh-attr-value">"a sh"</span>&gt;
    &lt;<span class="sh-prefix">xsl</span>:<span class="sh-element">import</span> <span class="sh-attr">href</span>=<span class="sh-attr-value">"../xsl/syntaxhighlight-xml.xsl"</span>/&gt;

    &lt;<span class="sh-prefix">xsl</span>:<span class="sh-element">template</span> <span class="sh-attr">match</span>=<span class="sh-attr-value">"/"</span>&gt;
      &lt;<span class="sh-element">pre</span>&gt;
        &lt;<span class="sh-prefix">xsl</span>:<span class="sh-element">apply-templates</span> <span class="sh-attr">mode</span>=<span class="sh-attr-value">"sh:xml"</span>/&gt;
      &lt;/<span class="sh-element">pre</span>&gt;
    &lt;/<span class="sh-prefix">xsl</span>:<span class="sh-element">template</span>&gt;

    &lt;<span class="sh-prefix">xsl</span>:<span class="sh-element">template</span> <span class="sh-attr">name</span>=<span class="sh-attr-value">"foo"</span>&gt;
      &lt;<span class="sh-prefix">xsl</span>:<span class="sh-element">text</span>&gt;<span class="sh-text">This text
will not
be indented.</span>&lt;/<span class="sh-prefix">xsl</span>:<span class="sh-element">text</span>&gt;
    &lt;/<span class="sh-prefix">xsl</span>:<span class="sh-element">template</span>&gt;
  &lt;/<span class="sh-prefix">xsl</span>:<span class="sh-element">stylesheet</span>&gt;<span class="sh-comment">

  &lt;!-- comment --&gt;
</span>
  &lt;<span class="sh-element">foo</span>
      <span class="sh-attr">bar</span>=<span class="sh-attr-value">"long value which makes the total length exceed 60 characters"</span>/&gt;
  &lt;<span class="sh-element">foo</span>&gt;<span class="sh-text">Short inline text.</span>&lt;/<span class="sh-element">foo</span>&gt;

  &lt;<span class="sh-element">foo</span>&gt;
    <span class="sh-text">Lorem ipsum dolor sit amet, consetetur sadipscing elitr.</span>
  &lt;/<span class="sh-element">foo</span>&gt;

  &lt;<span class="sh-element">bar</span>
      <span class="sh-attr">content</span>=<span class="sh-attr-value">"elements without child nodes are not surrounded by empty lines"</span>/&gt;
  &lt;<span class="sh-element">baz</span>
      <span class="sh-attr">content</span>=<span class="sh-attr-value">"elements without child nodes are not surrounded by empty lines"</span>/&gt;

  &lt;<span class="sh-prefix">xsd</span>:<span class="sh-element">annotation</span>&gt;
    &lt;<span class="sh-prefix">xsd</span>:<span class="sh-element">documentation</span>
        <span class="sh-xmlns-prefix">xmlns</span>=<span class="sh-namespace-uri">"<a href="http://www.w3.org/1999/xhtml">http://www.w3.org/1999/xhtml</a>"</span>&gt;
      &lt;<span class="sh-element">p</span>&gt;<span class="sh-text">Lorem ipsum</span>&lt;/<span class="sh-element">p</span>&gt;
    &lt;/<span class="sh-prefix">xsd</span>:<span class="sh-element">documentation</span>&gt;
  &lt;/<span class="sh-prefix">xsd</span>:<span class="sh-element">annotation</span>&gt;
&lt;/<span class="sh-prefix">a</span>:<span class="sh-element">doc</span>&gt;</pre>
